### PR TITLE
coprocessor: reduce the slow log count by changing the req_lifetime to total_process_time (#10858)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #10858 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tikv repo:
git pr https://github.com/tikv/tikv/pull/10863
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tikv.git pr/10863:release-4.0-e0c68a703674
```

---

### What problem does this PR solve?

#10841 *:Logger thread can be bottleneck and slow down the whole TiKV instance 

Problem Summary:

### What is changed and how it works?

Proposal: #10841

What's Changed:

### Related changes

- Need to cherry-pick to the release branch
release-3.0
release-3.1
release-4.0
release-5.0
release-5.1
release-5.2

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Slow log will only consider time spent on processing the request. 
- Drop log instead of blocking threads when slogger thread is overloaded and queue is filled up. 

### Release note 

```release-note
- TiKV coprocessor slow log will only consider time spent on processing the request. 
- Drop log instead of blocking threads when slogger thread is overloaded and queue is filled up. 
```

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
@breeswish Please help the contributor to resolve the conflict.

File: src/coprocessor/tracker.rs [hunk 1]
```diff
@@ -232,9 +232,17 @@ impl Tracker {
...
         if self.req_lifetime > self.slow_log_threshold {
             let some_table_id = self.req_ctx.first_range.as_ref().map(|range| {
                 tidb_query::codec::table::decode_table_id(range.get_start()).unwrap_or_default()
+=======
+        if self.total_process_time > self.slow_log_threshold {
+            let first_range = self.req_ctx.ranges.first();
+            let some_table_id = first_range.as_ref().map(|range| {
+                tidb_query_datatype::codec::table::decode_table_id(range.get_start())
+                    .unwrap_or_default()
+>>>>>>> e0c68a703... coprocessor: reduce the slow log count by changing the req_lifetime to total_process_time (#10858)
```
Comment (breeswish) [inline src/coprocessor/tracker.rs:245]:
```suggestion
        if self.total_process_time > self.slow_log_threshold {
            let some_table_id = self.req_ctx.first_range.as_ref().map(|range| {
                tidb_query::codec::table::decode_table_id(range.get_start()).unwrap_or_default()
```

**Review by skyzh — APPROVED** [review]

**Review by sunxiaoguang — APPROVED** [review]

**Review by breeswish — APPROVED** [review]
Thanks a lot!

**Pull request merged**